doc/linter_rules.md

```diff
@@ -0,0 +1,387 @@
+# Azure CLI Linter Rule Summary #
+This document lists the set of linter rules that would be applied to each PR in CI system.
+
+## Index
+* [Severity](#severity)
+* [Linter Rules](#linter-rules)
+* [Contributions](#contributions)
+* [Rule Descriptions](#rule-descriptions)
+
+## Severity
+Linter rule has three kinds of severity: High, Medium and Low.
+
+Rule with **high** severity is mandatory and must be passed in a PR. If you want bypass this linter rule, please contact corresponding reviewer.
+
+Rule with **medium** severity is not mandatory for a PR. But we recommend author following the guidance since they are best practice for Azure CLI.
```

Comment:
is there way to suppress high or medium rules? Will medium rules cause CI failure? high will definitely cause a task failure

Reply:
Yes. We can suppress any level rules in `linter_exclusions.yml` file. But I didn't put the method in the doc since I don't want everyone know how to suppress it. I think it's better to put a doc in our OneNote.

Currently, CI only run rules with high severity. Default severity is high. Let's have a discussion offline.